<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>

  <arg name="visualizer_frame" default="map"/>
  <arg name="scene_graph" doc="scene graph filename to show"/>
  <arg name="scene_graph_dir" default="$(find hydra)/output" doc="directory of scene graph file"/>
  <arg name="scene_graph_path" value="$(arg scene_graph_dir)/$(arg scene_graph)" doc="full path to scene graph to show"/>
  <arg name="publish_view_transforms" default="false" doc="publish top-down and orthographic views"/>
  <arg name="visualizer_ns" value="hydra_visualizer" doc="visualizer namespace"/>
  <arg name="start_rviz" default="true" doc="automatically start rviz"/>
  <arg name="verbosity" default="0" doc="visualizer verbosity"/>

  <arg name="mesh_filepath" default=""/>

  <include file="$(find hydra_ros)/launch/visualizer/includes/hydra_visualizer_params.xml"
           pass_all_args="true"/>

  <include file="$(find hydra_ros)/launch/visualizer/includes/hydra_visualizer_rviz.xml"
           pass_all_args="true"
           if="$(arg start_rviz)"/>

  <arg name="viz_debug" default="false"/>
  <arg name="viz_launch_prefix" value="gdb -ex run --args" if="$(arg viz_debug)"/>
  <arg name="viz_launch_prefix" value="" unless="$(arg viz_debug)"/>

  <node pkg="hydra_ros" type="hydra_visualizer_node" name="$(arg visualizer_ns)"
        launch-prefix="$(arg viz_launch_prefix)"
        output="screen"
        required="true"
        args="-alsologtostderr -colorlogtostderr -v=$(arg verbosity)">
    <param name="scene_graph_filepath" value="$(arg scene_graph_path)"/>
    <param name="load_graph" value="true"/>
  </node>

  <group if="$(arg publish_view_transforms)">
    <!-- Publish a static tf to visualize scene-graph in orthographic projection sideways -->
    <node pkg="tf2_ros"
        type="static_transform_publisher"
        name="ortho_tf_publisher"
        args="0 0 0 0.707 0 0 0.707 $(arg visualizer_frame) rviz_ortho"/>

    <!-- Publish a static tf to visualize scene-graph top-down -->
    <node pkg="tf2_ros"
        type="static_transform_publisher"
        name="topdown_tf_publisher"
        args="0 0 0 0 0.707 0 0.707 $(arg visualizer_frame) rviz_topdown"/>
  </group>

</launch>
